azure/azure.go

```
@@ -326,6 +326,57 @@ func (a *azureOps) DeleteFrom(diskName, _ string) error {
 	return a.Delete(diskName)
 }
 
+func (a *azureOps) Expand(
+	diskName string,
+	newSizeInGiB uint64,
+) (uint64, error) {
+	disk, err := a.disksClient.Get(
+		context.Background(),
+		a.resourceGroupName,
+		diskName,
+	)
+	if err != nil {
+		return 0, err
+	}
+
+	if disk.DiskProperties == nil || disk.DiskProperties.DiskSizeGB == nil {
+		return 0, fmt.Errorf("disk properties of (%v) is nil", diskName)
+	}
+
+	if *disk.DiskProperties.DiskSizeGB >= int32(newSizeInGiB) {
+		// no work is needed
```

same as AWS. The API should return an error here.